### ISSUE TITLE
Midpoint Review - Accessibility Feedback - GI Bill Comparison Tool, Specialized Mission Filter

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#64905]
## VFS actions
- [ ] Assign this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [ ] Comment on this ticket:
  - [ ] If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - [ ] When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - [ ] When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [x] Close the ticket when all feedback has been addressed.

## Thoughts/questions

- Sorry for the delayed feedback! Apparently my first attempt at creating this ticket didn't work, not sure what happened between me and GitHub on Friday.
- I noticed the staging link in your conversation guide and looked at a couple of things from that page, even though reviewing the coded product is outside the scope of the Midpoint Review. My apologies if I flagged anything that was a work-in-progress and wasn't ready for us yet!

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

- **Should:** Echoing Allison's comments in the meeting, definitely listen for feedback during your usability sessions about any potential confusion between the search button and the filter button. I think it should be fine, but I'm not one of your users! My concern here is cognitive load --- whether users get confused about what each button does, and whether that interferes with them completing tasks.
- **Should:** Echoing Erin's comments in #64826, I think some additional heading structure would be valuable here for indicating that the filters are a group together separate from the main search interaction. I think a heading to indicate the group of filters would make sense for all users, but if it's not practical or desirable for the visual design then a screen reader only `h2` would be great here.
- **Must:** In [staging](https://staging.va.gov/education/gi-bill-comparison-tool), each filter group is indicated by an `h5` heading, which doesn't follow the expected heading hierarchy (increments by more than 1 from the previous heading). I also don't think headings are the right element here. I'd recommend grouping each filter group as a `fieldset` and making the heading a `legend`. For screen reader users, this ensures that a user tabbing to the checkboxes will here the context for what they're interacting with, eg. their screen reader will announce "School types, Public" instead of a no-context "Public."
- **Must:** More heading stuff from staging: The [accordion component](https://design.va.gov/storybook/?path=/docs/components-va-accordion--default) supports setting a heading level, and I would expect the "Learn more about specialized missions" to be either an `h2` or an `h3`, depending on where the overall heading structure of the page lands. From there, all of the headings inside of the accordion should be one level higher than the accordion that contains them, eg. if the accordion is an `h2` then all of the headings in the accordion contents should be `h3`s. Note that it's totally okay for the headings to be styled to look different, like an `h3` styled to look like an `h6` for visual design purposes. But the heading level semantics are important for assistive technology users so the coded heading levels need to be reflective of the page structure.
- **Must:** With very few exceptions, we try to maintain material honesty --- in this case meaning, links look like links and buttons look like buttons. Since the Clear Filters button is coded as a button and performs an action (isn't navigational), it should be styled as a button, probably the [secondary button style](https://design.va.gov/components/button/#default---secondary). Likewise, the "Learn more" elements are coded as buttons and open modals, so those should also be styled as buttons as well. That's likely to be a bigger design challenge, and I wonder if it might be possible for those to be actual links that jump to an accordion at the bottom similar to the specialized mission filters?
- **Must:** A few more items on those Learn more button/links:
   - As currently coded, they seem to have very narrow click targets in the Y-axis --- when I try clicking, I'm seeing a click target narrower than the text itself --- which makes it likely for someone to miss them and accidentally check the box instead (thinking especially about users with a disability that impacts fine motor control). I'm wondering if this might be happening because they're nested inside the `label` element? Would it be possible to move those to their own line beneath the input label, and no longer nested inside the label?
   - Erin mentioned this as well, but the Learn more text should be more specific to indicate what you'll be learning more about. If it's not possible to have visible text with that level of specificity, then an `aria-label` is a must, eg. `aria-label="Learn more about accreditation"`. This will help screen reader users to differentiate between each element. As currently coded, both of them have an `aria-label` of "Learn more about VA education and training programs" which doesn't provide any additional specificity and doesn't help screen reader users differentiate between the two elements.
